### PR TITLE
Update the CMSIS-NN glue code to support int16 for softmax.

### LINKED ISSUE #43543: Add int16 support to the CMSIS-NN version of softmax
@tensorflow/micro

The softmax reference kernel have been updated with int16 support, and the CMSIS-NN version should to updated with int16 support as well.

### PR DESCRIPTION
**Pull request by jenselofsson**
Add int16 support to the CMSIS-NN version of softmax.

The changes are along the same lines as what was done for the reference kernel with #43320.

Fixes #43543

### CODE REVIEW

**Review by freddan80 — APPROVED** [review]
Looks good

**Comment by freddan80** [comment]
@njeffrie does it look ok to you?

**Comment by advaitjain** [comment]
This is mostly copied over from the changes in the reference kernel from https://github.com/tensorflow/tensorflow/pull/43320/, right? Can you mention that in the pull request description? And do mention any notable differences from the reference implementation.

We would like to change how the optimized kernels are implemented to allow for better code sharing but don't have a timeline for that. So let's update the commit description, resolve any conflicts and then we can get this merged.

**Comment by gbaned** [comment]
@jenselofsson  Can you please check @advaitjain's comments and keep us posted ? Thanks!

**Comment by mansnils** [comment]
This is partly fixing: https://github.com/tensorflow/tensorflow/issues/44455

**Review by advaitjain — APPROVED** [review]
